src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1577,9 +1577,7 @@ void enableSSL(String host,
         String tmfDefaultAlgorithm = null;  // Default algorithm (typically X.509) used by the TrustManagerFactory
         SSLHandhsakeState handshakeState = SSLHandhsakeState.SSL_HANDHSAKE_NOT_STARTED;
 
-        boolean isFips = false;
         String trustStoreType = null;
-        String fipsProvider = null;
```

Comment:
After further investigation, I figured out the root cause of the issue, 
Provider `IBMJCEFIPS` does not contain a keystore. The users should use non-IBMJCEFIPS keystores to store the keys. See [this page](https://www.ibm.com/support/knowledgecenter/en/SSYKE2_7.0.0/com.ibm.java.security.component.70.doc/security-component/progpractices.html). This means, the provider that contains keystore is not necessarily the same as the provider that contains FIPS. Below is a sample app that would work for IBM users with the current implementation.
```
ds.setTrustStoreType("JCEKS");
ds.setFIPS(true);
ds.setFIPSProvider("IBMJCE");  //Not IBMJCEFIPS
connection = (SQLServerConnection) ds.getConnection();
```
The suggestion above does not require any breaking changes, but the connection property `fipsProvider` has a confusing name.